src/uopool/memory.rs

```diff
@@ -0,0 +1,352 @@
+use educe::Educe;
+use ethers::types::{Address, U256};
+use parking_lot::RwLock;
+use std::{
+    collections::{HashMap, HashSet},
+    sync::Arc,
+};
+
+use crate::types::user_operation::{UserOperation, UserOperationHash};
+
+use super::{Mempool, MempoolId};
+
+pub type UserOperationsBySender = HashMap<Address, HashSet<UserOperationHash>>;
+
+#[derive(Educe)]
+#[educe(Debug)]
+pub struct MemoryMempool {
+    user_operations: Arc<RwLock<HashMap<UserOperationHash, UserOperation>>>, // user_operation_hash -> user_operation
+    user_operations_by_entry_point: Arc<RwLock<HashMap<MempoolId, HashSet<UserOperationHash>>>>, // mempool_id -> user_operations
+    user_operations_by_sender: Arc<RwLock<HashMap<MempoolId, UserOperationsBySender>>>, // mempool_id -> sender -> user_operations
```

Comment:
I think in the current stage it is kind of over-designed to have several entry_point mempools.

I was thinking that it is more flexible to have a simple `HashMap<UserOperationHash, UserOperation>` mempool and it could be extended to multiple entrypoint mempools by wrapping the simple mempool.

Reply:
What about HashMap at the `UoPoolService` level:
`pools: ... HashMap<MempoolId, Mempool>`
?

Reply:
That sounds better.